src/borg/helpers/parseformat.py

```diff
@@ -950,6 +937,138 @@ def format_iso_time(self, key, item):
+        fake_item2 = Item(mode=0, path="", user="", group="", mtime=0, uid=0, gid=0)
+        fake_diff = ItemDiff("", fake_item1, fake_item2, iter([]), iter([]))
+        formatter = cls("")
+        keys = []
+        keys.extend(formatter.call_keys.keys())
+        keys.extend(formatter.get_item_data(fake_diff).keys())
+        return keys
+
+    def __init__(self, format, content_only=False):
+        static_keys = {}  # here could be stuff on repo level, above archive level
+        static_keys.update(self.FIXED_KEYS)
+        super().__init__(
+            partial_format(format or "{content}{link}{directory}{blkdev}{chrdev}{fifo} {path}{NL}", static_keys)
+        )
+        self.content_only = content_only
+        self.item_data = static_keys
+        self.format_keys = {f[1] for f in Formatter().parse(format)}
+        self.call_keys = {
+            "content": self.format_content,
+            "mode": self.format_mode,
+            "type": partial(self.format_mode, ft=True),
+            "owner": partial(self.format_owner),
+            "group": partial(self.format_owner, spec="group"),
+            "user": partial(self.format_owner, spec="user"),
+            "link": partial(self.format_other, "link"),
+            "directory": partial(self.format_other, "directory"),
+            "blkdev": partial(self.format_other, "blkdev"),
+            "chrdev": partial(self.format_other, "chrdev"),
+            "fifo": partial(self.format_other, "fifo"),
+            "mtime": partial(self.format_time, "mtime"),
+            "ctime": partial(self.format_time, "ctime"),
+            "isomtime": partial(self.format_iso_time, "mtime"),
+            "isoctime": partial(self.format_iso_time, "ctime"),
+        }
+        self.used_call_keys = set(self.call_keys) & self.format_keys
+
+    def get_item_data(self, item: "ItemDiff", jsonline=False) -> dict:
+        diff_data = {}
+        if self.content_only and all(key not in item.changes() for key in self.CONTENT):
+            return {key: "" for key in [*self.used_call_keys, "path", "change", *self.item_data.keys()]}
+        for key in self.used_call_keys:
+            diff_data[key] = self.call_keys[key](item)
+
+        change = []
+        for key in self.call_keys:
+            if key in ("isomtime", "isoctime"):
+                continue
+            change.append(self.call_keys[key](item))
+        diff_data["change"] = " ".join([v for v in change if v])
+        diff_data["path"] = item.path
+        diff_data.update(self.item_data)
+        return diff_data
+
+    def format_other(self, key, diff: "ItemDiff"):
+        if key in diff.changes():
+            return f"{diff.changes()[key].flag}".ljust(27)
+        return ""
+
+    def format_mode(self, diff: "ItemDiff", ft=False):
+        change = diff.type() if ft else diff.mode()
+        if change:
+            return f"[{change.origin['past']} -> {change.origin['current']}]"
```

Comment:
didn't I advise against using old/new? now you used past/current, which is pretty much the same problem.

1/2, left/right, ... might be better (but left can be newer than right, so any reference to time / relative age might be confusing).

Reply:
does "before/after" suitable?

Reply:
some other places in the code just use item1 and item2.

Reply:
diff1/diff2 ?